mirascope/core/base/tool.py

```diff
@@ -8,25 +8,26 @@
 import inspect
 from abc import abstractmethod
 from collections.abc import Callable
-from typing import Any, ClassVar, TypeVar, cast
+from typing import Any, ClassVar, Generic, TypeVar, cast
 
 from pydantic import BaseModel, ConfigDict
-from typing_extensions import Required, TypedDict
+from typing_extensions import TypedDict
 
 from . import _utils
 
 _BaseToolT = TypeVar("_BaseToolT")
+_ToolSchema = TypeVar("_ToolSchema")
 
 
 class _CacheControl(TypedDict):
-    type: Required[str]
+    type: str
 
 
 class ToolConfig(TypedDict, total=False):
     cache_control: _CacheControl
 
 
-class BaseTool(BaseModel):
+class BaseTool(Generic[_ToolSchema], BaseModel):
```

Comment:
nit: I just ran the tests locally and got a ton of warnings, looks like the generic should come after `BaseModel`, i.e. `class BaseTool(BaseModel, Generic[_ToolSchema]): ...`

Reply:
Fixed